pennylane/qchem/convert.py

```diff
@@ -561,42 +562,47 @@ def _ucisd_state(cisd_solver, tol=1e-15):
 
 
 def import_state(solver, tol=1e-15):
-    r"""Convert an external wavefunction to a PennyLane state vector.
+    r"""Convert an external wavefunction to a state vector.
 
     Args:
         solver: external wavefunction object that will be converted
-        tol (float): the tolerance for discarding Slater determinants with small coefficients
+        tol (float): the tolerance for discarding Slater determinants based on their coefficients
 
     Raises:
-        ValueError: if ``method`` is not supported
+        ValueError: if external object type is not supported
 
     Returns:
-        statevector (array): normalized state vector of length 2**len(number_of_spinorbitals)
+        array: normalized state vector of length 2**len(number_of_spinorbitals)
 
     **Example**
 
     >>> from pyscf import gto, scf, ci
     >>> mol = gto.M(atom=[['H', (0, 0, 0)], ['H', (0,0,0.71)]], basis='sto6g')
     >>> myhf = scf.UHF(mol).run()
     >>> myci = ci.UCISD(myhf).run()
-    >>> wf_cisd = qml.qchem.convert.import_state(myci, tol=1e-1)
+    >>> wf_cisd = qml.qchem.import_state(myci, tol=1e-1)
     >>> print(wf_cisd)
     [ 0.        +0.j  0.        +0.j  0.        +0.j  0.1066467 +0.j
       0.        +0.j  0.        +0.j  0.        +0.j  0.        +0.j
       0.        +0.j  0.        +0.j  0.        +0.j  0.        +0.j
      -0.99429698+0.j  0.        +0.j  0.        +0.j  0.        +0.j]
     """
-    try:
-        import pyscf
-    except ImportError as Error:
-        raise ImportError(
-            "This feature requires pyscf. It can be installed with: pip install pyscf"
-        ) from Error
 
-    if isinstance(solver, pyscf.ci.ucisd.UCISD):
+    method = str(solver.__str__)
+
+    if "CISD" in method and "UCISD" not in method:
+        wf_dict = _rcisd_state(solver, tol=tol)
+    elif "UCISD" in method:
         wf_dict = _ucisd_state(solver, tol=tol)
+    elif "CCSD" in method and "UCCSD" not in method:
+        wf_dict = _rccsd_state(solver, tol=tol)
+    elif "UCCSD" in method:
+        wf_dict = _uccsd_state(solver, tol=tol)
```

Comment:
Personally, this seems brittle to me, and won't accommodate future methods like DMRG and SHCI. I think it would be more robust and scalable to have the user pass in the string identifying the method (e.g. `method = 'rcisd'), instead of detecting it from the object. Especially since this is what `_wfdict_to_statevector` already takes as input.

Reply:
> Especially since this is what `_wfdict_to_statevector` already takes as input.

Could you elaborate on this?

Reply:
Sorry, my bad on that last part re: `_wfdict_to_statevector`, got this mixed up with one of the tests. Discussed the main point offline and noted that we will think about this further, as there are conflicting concerns re: PRD UI vs scalability with respect to other methods.